### DISCUSSION TITLE
viem compile error, no direct calling of the functions, no typescript errors

### DISCUSSION THREAD

**Discussion by BrennerSpear** [wevm/viem#1702] [Question]
when I run `npm run compile` (which just runs `tsc`), I'm getting 6 errors like this:

` node_modules/viem/_types/utils/formatters/transaction.d.ts:26:1502 - error TS2536: Type 'K_2' cannot be used to index type '{ [K_3 in keyof TransactionEIP2930<bigint, number, boolean, "eip2930"> as K_3 extends keyof TOverrideReturnType ? TOverrideReturnType[K_3] extends void ? never : K_3 : K_3]: K_3 extends keyof TOverrideReturnType ? TOverrideReturnType[K_3] : TransactionEIP2930<...>[K_3]; } & TOverrideReturnType'.`

Relevant files:

package.json
```json
{
  "name": "web3-js-test",
  "version": "0.0.0",
  "description": "",
  "main": "build/src/index.js",
  "types": "build/src/index.d.ts",
  "engines": {
    "node": ">=10.4.0"
  },
  "files": [
    "build/src"
  ],
  "license": "Apache-2.0",
  "keywords": [],
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "lint": "gts lint",
    "local": "RPC_URL=http://localhost:8080/ethereum/mainnet/v1/11111111-1111-4111-9111-111111111111/v0 node build/src/index.js",
    "clean": "gts clean",
    "compile": "tsc",
    "fix": "gts fix",
    "pretest": "npm run compile",
    "posttest": "npm run lint"
  },
  "devDependencies": {
    "@types/node": "20.8.2",
    "concurrently": "^8.2.2",
    "gts": "^5.2.0",
    "nodemon": "^3.0.2",
    "typescript": "~5.2.0"
  },
  "dependencies": {
    "@google-cloud/functions-framework": "^3.3.0",
    "@google-cloud/pubsub": "^4.1.1",
    "viem": "^2.1.1"
  }
}

```

index.ts (the only typescript file)
```typescript
import functions from '@google-cloud/functions-framework';

import {createPublicClient, http, getFunctionSelector} from 'viem';
import {mainnet} from 'viem/chains';

...


```
<img width="1061" alt="Screenshot 2024-01-16 at 5 20 39 PM" src="https://github.com/wevm/viem/assets/12127609/d9ef8df3-8f14-42c0-a513-740baa12710e">

Any ideas?

**Comment by jxom** [comment]
Do you have a minimal reproducible example I can take a look at?

**Reply by BrennerSpear** [reply]
Shared a repo with you!

**Comment by FbartoliSyg** [comment]
Hey I do have the same issue. just installed the package and running tsc.
`blockchain-api-blockchain-1  | $ export $(cat .env) && nest start --watch --preserveWatchOutput
blockchain-api-blockchain-1  | [9:49:51 AM] Starting compilation in watch mode...
blockchain-api-blockchain-1  | 
blockchain-api-blockchain-1  | node_modules/viem/_types/utils/abi/parseEventLogs.d.ts:16:308 - error TS2344: Type 'eventName extends ContractEventName<abi>[] ? eventName[number] : eventName' does not satisfy the constraint 'ContractEventName<abi>'.
blockchain-api-blockchain-1  |   Type 'eventName | eventName[number]' is not assignable to type 'ContractEventName<abi>'.
blockchain-api-blockchain-1  |     Type 'eventName' is not assignable to type 'ContractEventName<abi>'.
blockchain-api-blockchain-1  |       Type 'ContractEventName<abi> | ContractEventName<abi>[]' is not assignable to type 'ContractEventName<abi>'.
blockchain-api-blockchain-1  |         Type 'ContractEventName<abi>[]' is not assignable to type 'ContractEventName<abi>'.
blockchain-api-blockchain-1  | 
blockchain-api-blockchain-1  | 16 export type ParseEventLogsReturnType<abi extends Abi | readonly unknown[] = Abi, eventName extends ContractEventName<abi> | ContractEventName<abi>[] | undefined = ContractEventName<abi>, strict extends boolean | undefined = boolean | undefined, derivedEventName extends ContractEventName<abi> | undefined = eventName extends ContractEventName<abi>[] ? eventName[number] : eventName> = Log<bigint, number, false, undefined, strict, abi, derivedEventName>[];
blockchain-api-blockchain-1  |                                                                                                                                                                                                                                                                                                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
blockchain-api-blockchain-1  | 
blockchain-api-blockchain-1  | node_modules/viem/_types/utils/formatters/transaction.d.ts:26:576 - error TS2536: Type 'K' cannot be used to index type '{ [K_1 in keyof TransactionLegacy<bigint, number, boolean, "legacy"> as K_1 extends keyof TOverrideReturnType ? TOverrideReturnType[K_1] extends void ? never : K_1 : K_1]: K_1 extends keyof TOverrideReturnType ? TOverrideReturnType[K_1] : TransactionLegacy<...>[K_1]; } & TOverrideReturnType'.
blockchain-api-blockchain-1  | 
blockchain-api-blockchain-1  | 26     format: (args: import("../../types/utils.js").Assign<Partial<RpcTransaction>, TOverrideParameters>) => (((import("../../types/transaction.js").TransactionLegacy<bigint, number, boolean, "legacy"> extends infer T_2 ? { [K_1 in keyof T_2 as K_1 extends keyof TOverrideReturnType ? TOverrideReturnType[K_1] extends void ? never : K_1 : K_1]: K_1 extends keyof TOverrideReturnType ? TOverrideReturnType[K_1] : import("../../types/transaction.js").TransactionLegacy<bigint, number, boolean, "legacy">[K_1]; } : never) & TOverrideReturnType extends infer T ? { [K in keyof T]: ((import("../../types/transaction.js").TransactionLegacy<bigint, number, boolean, "legacy"> extends infer T_1 ? { [K_1 in keyof T_1 as K_1 extends keyof TOverrideReturnType ? TOverrideReturnType[K_1] extends void ? never : K_1 : K_1]: K_1 extends keyof TOverrideReturnType ? TOverrideReturnType[K_1] : import("../../types/transaction.js").TransactionLegacy<bigint, number, boolean, "legacy">[K_1]; } : never) & TOverrideReturnType)[K]; } : never) | ((import("../../types/transaction.js").TransactionEIP2930<bigint, number, boolean, "eip2930"> extends infer T_5 ? { [K_3 in keyof T_5 as K_3 extends keyof TOverrideReturnType ? TOverrideReturnType[K_3] extends void ? never : K_3 : K_3]: K_3 extends keyof TOverrideReturnType ? TOverrideReturnType[K_3] : import("../../types/transaction.js").TransactionEIP2930<bigint, number, boolean, "eip2930">[K_3]; } : never) & TOverrideReturnType extends infer T_3 ? { [K_2 in keyof T_3]: ((import("../../types/transaction.js").TransactionEIP2930<bigint, number, boolean, "eip2930"> extends infer T_4 ? { [K_3 in keyof T_4 as K_3 extends keyof TOverrideReturnType ? TOverrideReturnType[K_3] extends void ? never : K_3 : K_3]: K_3 extends keyof TOverrideReturnType ? TOverrideReturnType[K_3] : import("../../types/transaction.js").TransactionEIP2930<bigint, number, boolean, "eip2930">[K_3]; } : never) & TOverrideReturnType)[K_2]; } : never) | ((import("../../types/transaction.js").TransactionEIP1559<bigint, number, boolean, "eip1559"> extends infer T_8 ? { [K_5 in keyof T_8 as K_5 extends keyof TOverrideReturnType ? TOverrideReturnType[K_5] extends void ? never : K_5 : K_5]: K_5 extends keyof TOverrideReturnType ? TOverrideReturnType[K_5] : import("../../types/transaction.js").TransactionEIP1559<bigint, number, boolean, "eip1559">[K_5]; } : never) & TOverrideReturnType extends infer T_6 ? { [K_4 in keyof T_6]: ((import("../../types/transaction.js").TransactionEIP1559<bigint, number, boolean, "eip1559"> extends infer T_7 ? { [K_5 in keyof T_7 as K_5 extends keyof TOverrideReturnType ? TOverrideReturnType[K_5] extends void ? never : K_5 : K_5]: K_5 extends keyof TOverrideReturnType ? TOverrideReturnType[K_5] : import("../../types/transaction.js").TransactionEIP1559<bigint, number, boolean, "eip1559">[K_5]; } : never) & TOverrideReturnType)[K_4]; } : never)) & { [_key in TExclude[number]]: never; };
blockchain-api-blockchain-1  |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           `

**Reply by FbartoliSyg** [reply]
Version : 
- node v20.9.0
- typescript 5.1.6
- viem: 2.2.0

I donwgraded viem to 2.1.1 and I got the same issue

**Reply by FbartoliSyg** [reply]
For now I am hidding the error with tsconfig : skipLibCheck true and it does work as expected

**Comment by pepoospina** [comment]
I have the same issue here (15 errors)

my `package.json`

```
{
  "name": "functions",
  "scripts": {
    "lint": "eslint --ext .js,.ts .",
    "build": "tsc",
    "build:watch": "tsc --watch",
    "serve": "npm run build && firebase emulators:start --only functions",
    "shell": "npm run build && firebase functions:shell",
    "start": "npm run shell",
    "deploy": "firebase deploy --only functions",
    "logs": "firebase functions:log"
  },
  "engines": {
    "node": "18"
  },
  "main": "lib/index.js",
  "dependencies": {
    "@nanopub/sign": "^0.0.18",
    "@types/n3": "^1.16.4",
    "@types/node-forge": "^1.3.11",
    "@types/pino": "^7.0.5",
    "cors": "^2.8.5",
    "crypto-js": "^4.2.0",
    "express": "^4.18.2",
    "firebase-admin": "^11.8.0",
    "firebase-functions": "^4.3.1",
    "jsonwebtoken": "^9.0.2",
    "n3": "^1.17.2",
    "node-fetch": "^3.3.2",
    "node-forge": "^1.3.1",
    "oauth-1.0a": "^2.2.6",
    "pino": "^8.17.2",
    "pino-pretty": "^10.3.1",
    "twitter-api-v2": "^1.16.0",
    "viem": "^2.7.10",
    "yup": "^1.3.3"
  },
  "devDependencies": {
    "@trivago/prettier-plugin-sort-imports": "^4.3.0",
    "@types/chai": "^4.3.4",
    "@types/mocha": "^10.0.1",
    "@types/supertest": "^6.0.2",
    "@typescript-eslint/eslint-plugin": "^5.12.0",
    "@typescript-eslint/parser": "^5.12.0",
    "chai": "^4.3.7",
    "firebase-functions-test": "^3.1.0",
    "mocha": "^10.2.0",
    "supertest": "^6.3.4",
    "ts-node": "^10.9.1",
    "typescript": "^5.3.3"
  },
  "private": true
}
```

**Reply by FbartoliSyg** [reply]
For now in your TS file set skipLibCheck to true

**Comment by pepoospina** [comment]
I just had to still set skipLibCheck to true. Just in case.
